cmd/commands/datasetArchiver.go

```diff
@@ -1,143 +1,130 @@
-/*
-
-This script archives all datasets in state datasetCreated from a given ownerGroup
-
-*/
-
-package main
+package cmd
 
 import (
 	"bufio"
 	"crypto/tls"
-	"flag"
 	"fmt"
 	"log"
 	"net/http"
 	"os"
 	"strings"
 	"time"
 
-	"github.com/paulscherrerinstitute/scicat/datasetUtils"
-
 	"github.com/fatih/color"
+	"github.com/paulscherrerinstitute/scicat/datasetUtils"
+	"github.com/spf13/cobra"
 )
 
-var VERSION string
-
-func main() {
-	var client = &http.Client{
-		Transport: &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: false}},
-		Timeout:   10 * time.Second}
-
-	const PROD_API_SERVER string = "https://dacat.psi.ch/api/v3"
-	const TEST_API_SERVER string = "https://dacat-qa.psi.ch/api/v3"
-	const DEV_API_SERVER string = "https://dacat-development.psi.ch/api/v3"
-	const LOCAL_API_SERVER string = "http://localhost:3000/api/v3"
-
-	const MANUAL string = "http://melanie.gitpages.psi.ch/SciCatPages"
-	const APP = "datasetArchiver"
-	var scanner = bufio.NewScanner(os.Stdin)
-
-	var APIServer string
-	var env string
-
-	// pass parameters
-	userpass := flag.String("user", "", "Defines optional username and password")
-	token := flag.String("token", "", "Defines optional API token instead of username:password")
-	tapecopies := flag.Int("tapecopies", 1, "Number of tapecopies to be used for archiving")
-	testenvFlag := flag.Bool("testenv", false, "Use test environment (qa) instead or production")
-	localenvFlag := flag.Bool("localenv", false, "Use local environment (local) instead or production")
-	devenvFlag := flag.Bool("devenv", false, "Use development environment instead or production")
-	nonInteractiveFlag := flag.Bool("noninteractive", false, "Defines if no questions will be asked, just do it - make sure you know what you are doing")
-	showVersion := flag.Bool("version", false, "Show version number and exit")
-
-	flag.Parse()
-
-	if datasetUtils.TestFlags != nil {
```

Comment:
was this block removed by mistake?

Reply:
having this will allow to check that passing variables by reference did not introduce any error

Reply:
it was deleted because the code was original only copied from before I added tests, this is readded in the latest commit.